src/AutoRest.CSharp/Common/Output/Expressions/Snippets.New.cs

```diff
@@ -50,7 +50,7 @@ public static RequestContextExpression RequestContext(CancellationTokenExpressio
 
             public static ValueExpression RequestFailedException(BaseResponseExpression response) => Instance(Configuration.ApiTypes.RequestFailedExceptionType, response);
 
-            public static ResourceIdentifierExpression ResourceIdentifier(ValueExpression resourceData) => new(Instance(typeof(ResourceIdentifier), new MemberExpression(resourceData, "Id")));
+            public static ResourceIdentifierExpression ResourceIdentifier(ValueExpression idString) => new(Instance(typeof(ResourceIdentifier), idString));
```

Comment:
This method is only used in a scenario that we are meant to create an expression of `new ResourceIdentifier(idString)`, the previous implemenation is wrong and misleading